### PR TITLE
Fix gateway ping (BugFix)

### PR DESCRIPTION
**Pull request by kissiel**
## Description

This PR simplifies the argument handling in `gateway_ping_test` test script.

The `count`, `threshold`, and `deadline` options were never used and they made the logic complicated. The verbosity option was also a mixed bag, and [lately was suffering from a bug](https://github.com/canonical/checkbox/issues/993)

~~This is a partial fix for the [#371](https://github.com/canonical/checkbox/issues/371) bug.~~

Namely it moves code around to make allow for getting the interface name.
This PR doesn't change anything, just adds an command-line option and refactors some functions

## Resolved issues

Fixes: 
https://github.com/canonical/checkbox/issues/993
https://warthogs.atlassian.net/browse/CHECKBOX-1258

https://github.com/canonical/checkbox/issues/992

## Tests

Unit tests were added and the previous ones got adjusted.

### CODE REVIEW

**Review by Hook25 — DISMISSED** [review]
+1, lgtm

**Comment by kissiel** [comment]
After testing it seems that this PR also fixes https://github.com/canonical/checkbox/issues/992

File: providers/base/bin/gateway_ping_test.py [hunk 1]
```diff
@@ -474,52 +400,72 @@ def main(argv) -> int:
...
+
+    :param interface: the interface name to check
+    :return: True if the interface is a cable interface, False otherwise
+
+    Looking at the `man 7 systemd.net-naming-scheme` we can see that
+    even the `eth` matching may be an overkill.
+    """
+    if not isinstance(interface, str) or not interface:
+        return False
+    return interface.startswith("en") or interface.startswith("eth")
```
Comment (Hook25) [inline providers/base/bin/gateway_ping_test.py:467]: It is a bit weird, given the `Route` structure, that these are free functions

File: providers/base/bin/gateway_ping_test.py [hunk 2]
```diff
@@ -474,52 +400,72 @@ def main(argv) -> int:
...
 
+def get_default_gateways() -> Dict[str, str]:
+    """
+    Use `route` program to find default gateways for all interfaces.
+
+    returns a dictionary in a form of {interface_name: gateway}
+    """
+    try:
+        routes = subprocess.check_output(
+            ["route", "-n"], universal_newlines=True
```
Comment (Hook25) [inline providers/base/bin/gateway_ping_test.py:442]: I didn't double check what it does to be honest, but before we were defining and envvar called `env={"LANGUAGE": "C"},`, is the omission of that intentional?

**Review by Hook25 — APPROVED** [review]
Well done, give a thought to the comment below, I would have used the api in Route to instanciate N routes and getting default gateways via that class. This just because the Route class has some robustness to it, trying to fetch the default gateway via multiple routes.

File: providers/base/bin/gateway_ping_test.py (hunk 2 above)
Comment (kissiel) [reply providers/base/bin/gateway_ping_test.py:442]: I did not remove it by accident, but I considered that we *never* strip the locale in other places, I'd rather:
* make it consistent
* **if** the need arises it'll be better to introduce like a `locale_less_check_output` that does this to all calls of `check_output`
* make it simpler

File: providers/base/bin/gateway_ping_test.py (hunk 1 above)
Comment (kissiel) [reply providers/base/bin/gateway_ping_test.py:467]: I don't know how I feel about using `class`es as namespaces. ATM, for me the less digging, the less indent, the better.

File: providers/base/bin/gateway_ping_test.py (hunk 2 above)
Comment (Hook25) [reply providers/base/bin/gateway_ping_test.py:442]: ok I wanted to make sure, we can go back to this if troubles arises

**Pull request merged**